### PR TITLE
NFT purchase improvements

### LINKED ISSUE #3070: NFT purchase improvements (price + fee)
**Scope:**

- [ ] Include price breakdown in Buy now flow ("you will pay" and "transacion fee")
- [ ] Show loading state for "You will pay" when the fee is being fetched
- [ ] Ensure fee loads properly for both buy now and place bid
- [ ] Ensure price + fee is properly validated

---

I think there is a little discrepancy between what we show in the `Place bid` view and what we show in the `Buy now` view. 
Yes - there is a fee in the action bar, but I think we should be more clear about how much the user needs to actually pay
Place bid view:
image
Buy now view
image

### PR DESCRIPTION
**Pull request by rafalpawlow**
Closes #3070 

**Design**
https://www.figma.com/file/wjeAOZq77pSUFZsYXJqKIB/NFT-purchase-view?node-id=1036%3A102823

### CODE REVIEW

**Review by kdembler — CHANGES_REQUESTED** [review]
There's a split second after you input the bid that the bid in summary is shown before the loading triggers:

https://user-images.githubusercontent.com/12646744/185422009-d5b99f07-de12-492c-be6a-8718f8971b39.mp4

File: packages/atlas/src/views/global/NftPurchaseBottomDrawer/NftPurchaseBottomDrawer.tsx [hunk 1]
```diff
@@ -587,27 +583,42 @@ export const NftPurchaseBottomDrawer: FC = () => {
...
+                  makeBidFeeLoading ? (
+                    <SkeletonLoader width={80} height={16} />
+                  ) : (
+                    <NumberFormat as="span" value={transactionFee} withToken variant="t100" color="colorText" />
+                  )}
                 </Row>
                 <Row>
                   <Text as="span" variant="h500" color="colorText">
                     You will pay
                   </Text>
```
Comment (drillprop) [inline packages/atlas/src/views/global/NftPurchaseBottomDrawer/NftPurchaseBottomDrawer.tsx:603]: In the design there is a short text: "You need to fill out the amount first" when the input is empty. You didn't include that. 
https://www.figma.com/file/wjeAOZq77pSUFZsYXJqKIB/NFT-purchase-view?node-id=1036%3A102849

**Review by drillprop — CHANGES_REQUESTED** [review]
I still don't like the flow and how validation works there, but not sure if the designs are up to date. 
1. When I click buy now and I don't have money. I still see the big call to action button "buy nft", which encourages me as a user to click: 
image
2.  In the inputs I know that at some point we decided that we will do validation on blur.  But should we actually do it that way in this case? Maybe we could use `onChange` mode in this case.  I'm still confused about what's happening to be honest

Can you consult this with a designer? I mean, I'm fine if we leave it as it is and if we decide to change that we could do this as a follow-up.

**Comment by toiletgranny** [comment]
@drillprop That's a fair point. The UX here is far from ideal, and the designs did not really provide a solid solution for that case. That's why I created [this issue](https://github.com/Joystream/atlas/issues/3084#issue-1329710328), and now that I think of it, @kdembler, I think we should reopen it and let me take a closer look and think about it. I'd love to give you guys some ideas right now, but I really have to get back to Pioneer (cc @dmtrjsg)...

If you agree, I'd leave this implementation in its current state (in the PR) and open up a new PR once we know how to handle this case properly.

**Comment by drillprop** [comment]
> @drillprop That's a fair point. The UX here is far from ideal, and the designs did not really provide a solid solution for that case. That's why I created [this issue](https://github.com/Joystream/atlas/issues/3084#issue-1329710328), and now that I think of it, @kdembler, I think we should reopen it and let me take a closer look and think about it. I'd love to give you guys some ideas right now, but I really have to get back to Pioneer (cc @dmtrjsg)...
> 
> If you agree, I'd leave this implementation in its current state (in the PR) and open up a new PR once we know how to handle this case properly.

Thanks for your input, I'd opt not to reopen your issue. Perhaps we could create another issue (third one😅) and rethink validation strategies in these particular cases - `buy now nft` and `placing bid`. And yeah, I'm also fine with doing this some other time. @kdembler wdyt?

**Review by drillprop — APPROVED** [review]
LGTM

**Review by kdembler — APPROVED** [review]

**Pull request merged**